src/broker/runat.c

```diff
@@ -339,9 +337,14 @@ static int runat_command_modenv (struct runat_command *cmd,
                                  const char *local_uri)
 {
     if (blocklist) {
-        int i;
-        for (i = 0; blocklist[i] != NULL; i++)
-            flux_cmd_unsetenv (cmd->cmd, blocklist[i]);
+        for (int i = 0; blocklist[i] != NULL; i++) {
+            if (strchr (blocklist[i], '?')
+                || strchr (blocklist[i], '*')
+                || strchr (blocklist[i], '['))
+                flux_cmd_unsetenv_glob (cmd->cmd, blocklist[i]);
+            else
+                flux_cmd_unsetenv (cmd->cmd, blocklist[i]);
```

Comment:
This makes me wonder if `flux_cmd_unsetenv()` should just optionally take a glob pattern and push detection of the pattern inside that function?

Also, elsewhere we have these 3 tests abstracted into an `isa_glob()` function. If it isn't quite enough to warrant moving to libutil at this point (and it is such a small function that I don't think so either), then perhaps using `isa_glob()` here (or in `flux_cmd_unsetenv()` would at least allow the common function to easily be discovered later...

Reply:
Oh duh, excellent idea.  I'll do that.